src/Time/Rational.hs

```diff
@@ -114,3 +127,17 @@ Normalize (9 % 12) :: Rat
 -}
 type family Normalize (r :: Rat) :: Rat  where
     Normalize (a :% b) = (a `Div` Gcd a b) :% (b `Div` Gcd a b)
+
+
+-- | Rational numbers, with numerator and denominator of 'Natural' type.
+type RatioNat = Ratio Natural
+
+-- | This class gives the integer associated with a type-level rational.
+class KnownRat (r :: Rat) where
+  ratVal :: Proxy r -> Ratio Natural
+
+instance (KnownNat a, KnownNat b) => KnownRat (a :% b) where
+    ratVal _ = natVal (Proxy :: Proxy a) :% natVal (Proxy :: Proxy b)
```

Comment:
If you add `TypeApplications` extension you can write this as `Proxy @a` and `Proxy @b`.